x-pack/plugins/security_solution/public/resolver/store/middleware/current_related_event_fetcher.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Dispatch, MiddlewareAPI } from 'redux';
+import { isEqual } from 'lodash';
+import { SafeResolverEvent } from '../../../../common/endpoint/types';
+
+import { ResolverState, DataAccessLayer, PanelViewAndParameters } from '../../types';
+import * as selectors from '../selectors';
+import { ResolverAction } from '../actions';
+import { eventIDSafeVersion } from '../../../../common/endpoint/models/event';
+
+/**
+ *
+ * @description - This api is called after every state change. If the current view is the `eventDetail` view
+ * it will first check state to see if we have the event in question, otherwise
+ * it will request the event details from the server.
+ * @export
+ * @param {DataAccessLayer} dataAccessLayer
+ * @param {MiddlewareAPI<Dispatch<ResolverAction>, ResolverState>} api
+ * @returns {() => void}
+ */
+export function CurrentRelatedEventFetcher(
+  dataAccessLayer: DataAccessLayer,
+  api: MiddlewareAPI<Dispatch<ResolverAction>, ResolverState>
+): () => void {
+  let last: PanelViewAndParameters | undefined;
+
+  return async () => {
+    const state = api.getState();
+
+    const newParams = selectors.panelViewAndParameters(state);
+
+    const oldParams = last;
+    last = newParams;
+
+    // If the panel view params have changed and the current panel view is the `eventDetail`, then fetch the event details for that eventID.
+    if (!isEqual(newParams, oldParams) && newParams.panelView === 'eventDetail') {
+      const currentEventID = newParams.panelParameters.eventID;
+      const nodeID = newParams.panelParameters.nodeID;
+      const fetchedRelatedEvents = state.data.relatedEvents.get(nodeID);
+      let existingEvent: SafeResolverEvent | undefined;
+
+      // Check if the event data already exists in state
+      if (fetchedRelatedEvents && fetchedRelatedEvents.events.length > 0) {
```

Comment:
I think we should omit this part. [My PR](https://github.com/elastic/kibana/pull/78780) decouples the 'nodeEventsInCategory' panel from this piece of state. The `relatedEvents` state will not be used otherwise and I think any type of caching should be out of scope for this story.

Reply:
Yea, I just saw that in your PR. Will remove it. Thanks